### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by 3Hren**
Actually not.

Changed service pool architecture. Also **everything** has been refactored. Note, that this is WIP.

### CODE REVIEW

**Comment by 3Hren** [comment]
@diunko @noxiouz @shindo @iderikon @antmat take a look.

File: src/server.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    request_timeout = 5;
+    if (config.HasMember("request_timeout")) {
+        request_timeout = config["request_timeout"].GetUint();
+    }
+
+    CP_INFO("proxy has successfully started");
+
+    on<handler::ping_t>   (options::prefix_match("/ping"));
```
Comment (noxiouz) [inline src/server.cpp]: It hides the ```/ping``` from applications.

Comment (3Hren) [inline src/server.cpp]: I don't know how to implement this functionality using thevoid without splitting headers and path handlers.

**Comment by kobolog** [comment]
Not that minor =) Stale PR, bump!

**Comment by 3Hren** [comment]
Who cares now, when we have https://github.com/3Hren/cocaine-http-proxy?

**Pull request closed without merging**